src/search/search.ts

```diff
@@ -0,0 +1,24 @@
+import { Product } from './product';
+
+/** The name of the event fired to request a search. */
+export const SEARCH_REQUEST = 'sfx::search_request';
+/** The type of the [[SEARCH_REQUEST]] event payload. */
+export interface SearchRequestPayload {
+ /** The ID of the search box that initiated this request. */
+ searchbox?: string;
+ /** The search term. */
+ query: string;
+}
+
+/** The name of the event fired when the results of a search request have been received.  */
+export const SEARCH_RESPONSE = 'sfx::search_response';
+/** The type of the [[SEARCH_RESPONSE]] event payload. */
+export interface SearchResponsePayload {
+  /** The products returned in the search response. */
```

Comment:
We may need an optional `searchbox` attribute here. If there are multiple `SAYT` (or `Products`) components, they'll need to know which one should populate its products.